source-code/core/package.json

```diff
@@ -30,7 +30,11 @@
   "scripts": {
     "build": "tsc --build ./tsconfig.build.json",
     "dev": "tsc --watch",
-    "test": "vitest run --passWithNoTests --coverage",
+    "test": "tsc && vitest",
+    "tsc": "tsc --noEmit",
+    "tsc:watch": "tsc --noEmit",
```

Comment:
Does build and dev not suffice?

Reply:
`dev` outputs the code.
`tsc` will just check for errors.
You can see `tsc` as the `vitest` command for `TypeScript`. It will make more sense if we implement `type-definition` tests (see [example here](https://github.com/ivanhofer/exceptionally/blob/main/src/core/utils.test.types.ts)). There the `tsconfig` for `build` and `test` will be different.